### DISCUSSION TITLE
Error using react-input-mask

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#9085] [Q&A] [answered]
I am trying to use react-input-mask as per the sandbox I found but it doesn't appear to work with newer versions of react hook form. This sandbox is showing version 5 I think and I am using version 7.27.1.

The error I get is _**props.render is not a function**_

https://codesandbox.io/s/react-hook-form-with-inputmask-forked-c7nl2o

```
                <Controller
                  as={InputMask}
                  control={control}
                  mask="999.999.999-99"
                  name="phone"
                />
```

**Comment by bluebill1049** [comment, marked as answer]
have a look at these examples: https://codesandbox.io/s/react-hook-form-v7-controller-5h1q5 they may helps.

**Reply by ghost** [reply]
Thanks Bill, after I posted my question I came across react-number-format which seems like a better fit as your example confirmed. Seems like I will need to use regex or Yup .matches because just checking for min length of 10 isn't working.
